wordpress-mobile/WordPress-iOS
# Jetpack Section: Restore Status and Restore Complete

Part of #15191

- Hooked up restore site APIs
    - Restore site
    - Get rewind status (poll for status **every 5 seconds**)
- Added a new Restore Complete screen

### Merge directions:
- ⚠️ Merge https://github.com/wordpress-mobile/WordPress-iOS/pull/15633 first

### Notes:
- Styling isn't 100% complete
- The `Visit site` button on the Restore Complete screen isn't hooked up yet
- Error handling will be addressed in a separate PR

### To test:

1. My Site > Activity Log > rewindable activity cell
2. Tap on ellipsis icon
3. Tap on `Restore`
4. Tap on `Restore to this point`
5. Tap on `Confirm`
     - ✅ The progress view should update
     - ✅ Once the site has been restored, the app should transition to the Restore Complete screen (Note that it might take a little while to transition after it hits `100%`)
     

30% (`.running`)  | 100% (`.running`) | Complete (`.finished`)
--- | --- | ---
![Screen Shot 2021-01-14 at 19 38 26](https://user-images.githubusercontent.com/6711616/104580236-57a32d80-56a0-11eb-8ad9-17c7771312db.png) | ![Screen Shot 2021-01-14 at 19 38 34](https://user-images.githubusercontent.com/6711616/104580247-5a058780-56a0-11eb-88b5-d19a5130f71f.png) | ![Screen Shot 2021-01-14 at 19 38 55](https://user-images.githubusercontent.com/6711616/104580251-5c67e180-56a0-11eb-8285-38aa93c96662.png) 

### PR submission checklist:
